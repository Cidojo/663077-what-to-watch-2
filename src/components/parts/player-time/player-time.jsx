import * as React from 'react';
import * as PropTypes from 'prop-types';

import {formatCurrentTime} from './../../../utils/utils.js';
import {evalProgress} from './../../../utils/utils.js';

const PlayerTime = (props) => {
  const {duration, currentTime} = props;
  const progress = evalProgress(currentTime, duration);


  return (
    <>
      <div className="player__time">
        <progress className="player__progress" value={progress} max="100"></progress>
        <div
          className="player__toggler"
          style={{
            left: `${progress}%`
          }}
        >Toggler</div>
      </div>
      <div className="player__time-value">{formatCurrentTime(currentTime)}</div>
    </>
  );
};

PlayerTime.propTypes = {
  duration: PropTypes.number,
  currentTime: PropTypes.number
};

PlayerTime.defaultProps = {
  duration: 0,
  currentTime: 0
};

export {PlayerTime};

