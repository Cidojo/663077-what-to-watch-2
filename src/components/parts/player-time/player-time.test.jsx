import * as React from 'react';
import {PlayerTime} from './player-time.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  duration: 0,
  currentTime: 0
};

it(`should render PlayerTime component without errors`, () => {
  const tree = renderer.create(<PlayerTime {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
