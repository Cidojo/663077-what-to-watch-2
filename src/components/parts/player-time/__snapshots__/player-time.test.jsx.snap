// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlayerTime component without errors 1`] = `
Array [
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={0}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      Toggler
    </div>
  </div>,
  <div
    className="player__time-value"
  >
    00:00:00
  </div>,
]
`;
